File: app/src/main/java/com/baidu/tv/player/geocoding/GeocodingStrategy.java
package com.baidu.tv.player.geocoding;

import android.content.Context;

/**
 * 逆地理编码策略接口
 * 定义了将GPS坐标转换为地址信息的标准接口
 */
public interface GeocodingStrategy {
    
    /**
     * 获取策略名称
     * @return 策略名称（如：AMap、Geocoder、Nominatim）
     */
    String getName();
    
    /**
     * 检查策略是否可用
     * @param context Android上下文
     * @return true表示策略可用，false表示不可用
     */
    boolean isAvailable(Context context);
    
    /**
     * 获取策略的优先级
     * 数值越小，优先级越高
     * @return 优先级值（建议范围：1-100）
     */
    int getPriority();
    
    /**
     * 根据GPS坐标获取地址
     * @param context Android上下文
     * @param latitude 纬度（WGS84坐标系）
     * @param longitude 经度（WGS84坐标系）
     * @return 地址字符串，失败返回null
     */
    String getAddress(Context context, double latitude, double longitude);
    
    /**
     * 获取策略的超时时间（毫秒）
     * @return 超时时间
     */
    int getTimeout();
    
    /**
     * 获取策略的描述信息
     * @return 描述信息
     */
    String getDescription();
}

File: app/src/main/java/com/baidu/tv/player/kt/location/geocoding/GeocodingStrategy.kt
package com.baidu.tv.player.kt.location.geocoding

import com.baidu.tv.player.kt.location.GpsCoordinate

/**
 * 逆地理编码策略接口（对应 tasks.md 7.8，迁移自 Java 版 GeocodingStrategy）。
 *
 * 将 WGS84 [GpsCoordinate] 反解为人类可读地址。所有实现必须：
 * - 为 suspend 函数，运行在 IO 调度器（由调用方切换）；
 * - 失败/无结果时静默返回 null，绝不抛出异常给上层；
 * - 不请求任何设备定位权限，仅根据传入坐标查询。
 */
interface GeocodingStrategy {

    /** 策略名称（用于日志/选择）。 */
    val name: String

    /**
     * 优先级，数值越小优先级越高。工厂按此排序进行回退。
     */
    val priority: Int

    /**
     * 当前环境下该策略是否可用（如 API key 是否配置、Geocoder 是否 present）。
     */
    fun isAvailable(): Boolean

    /**
     * 逆地理编码：返回格式化地址；失败或无结果返回 null。
     */
    suspend fun getAddress(coordinate: GpsCoordinate): String?
}
